practice/case.py

```diff
@@ -0,0 +1,55 @@
+""'This program converts text from snake_case to CamelCase and from CamelCase to snake_case""'
+
+# 1. Setup
+import re
+
+# 2. Define
+
+def prompt_for_input_phrase():
+    """Gathers input word from user"""
+    return input('What is the word you would like to transform? ')
+    # return 'test_word'
+
+def detect_case(input_word):
+    """Detects whether the word is in snake_case or not, generates bool value storing this
+    >>> detect_case('test_word')
+    True
+    """
+    is_snake_case = False if input_word[0].isupper() == True else True
+    return is_snake_case
+
+def get_word_list(is_snake_case_bool, input_phrase):
+    """This program inputs a bool identifying whether a phrase is in either snake_case or CamelCase, the phrase itself,
+     and then splits the phrase into a list of words based on which case is used.
+     >>> get_word_list(True, 'test_word')
+     ['test', 'word']
+     """
+    word_list = re.findall('[A-Z][^A-Z]*', input_phrase) if is_snake_case_bool == False else input_phrase.split('_')
```

Comment:
Converting `CamelCase` to a word list and converting `snake_case` to a word list are two separate steps that should be in two separate and testable functions.